tests/foreman/cli/test_usergroup.py

```diff
@@ -174,64 +172,60 @@ def test_positive_remove_user_assigned_to_usergroup(self):
         user = make_user()
         user_group = make_usergroup()
         UserGroup.add_user({'id': user_group['id'], 'user-id': user['id']})
-        with self.assertNotRaises(CLIReturnCodeError):
-            User.delete({'id': user['id']})
+        User.delete({'id': user['id']})
+        user_group = UserGroup.info({'id': user_group['id']})
+        assert user['login'] not in user_group['users']
 
 
 @pytest.mark.run_in_one_thread
-class ActiveDirectoryUserGroupTestCase(CLITestCase):
+class TestActiveDirectoryUserGroup:
     """Implements Active Directory feature tests for user groups in CLI."""
 
-    @classmethod
+    @skip_if_not_set('ipa')
+    @pytest.fixture(scope='module')
+    def module_creds(self):
+        ldap_user_name = settings.ldap.username
+        ldap_user_passwd = settings.ldap.password
+        return {'username': ldap_user_name, 'password': ldap_user_passwd}
+
     @skip_if_not_set('ldap')
-    def setUpClass(cls):
+    @pytest.fixture(scope='module')
+    def module_auth(self, module_creds):
```

Comment:
The scope might be more than you want to tackle in this conversion PR, but there is a `module_auth` fixture in both classes that could be consolidated to a single fixture function outside of the classes. It would take some direct/indirect parametrization to handle the AD/IPA differences.  Just a thought to improve things a bit.

Reply:
If you wanted to see an example here https://github.com/omkarkhatavkar/robottelo/blob/fa4e05ce3960bf5c4ba5a13b5da73bda551223b0/pytest_fixtures/satellite_auth.py#L97 it is. It has done for API.